### ISSUE TITLE
Is it suitable to achieve DensifyPointClould with CUDA?

### ISSUE THREAD

**Issue by numbmelon** [cdcseacave/openMVS#654]
I want to use GPU programming to optimize the running time of the Reconstruction of DensePointCloud.  May I ask if it's feasible to rewrite some functions in SceneDensify with CUDA to achieve that? If possible, could I directly use some kernel functions you wrote in the SceneRefineCUDA to reduce the amout of work? Could you give me some suggestions? Thanks in advance!

**Comment by cdcseacave** [comment]
yes, the patchmatch algorithm greatly benefits from a GPU implementation and for that `EstimateDepthMapTmp()` should be ported; however the code can not just be directly ported to CUDA, the propagation step (the way the depth-map image is parsed) needs to be changed from pseudo serial mode as it is now, to multithreded mode as required by a GPU; this can be achieved by using a checkerboard approach, however the propagation can suffer; this is well explained in `Xu and Tao - 2019 - Multi-Scale Geometric Consistency Guided Multi-Vie`

**Comment by cdcseacave** [comment]
Closing as stale. The codebase has been substantially updated on the \develop\ branch. Please reopen with reproduction steps if the issue persists on the latest code.
